src/TorchSharp/NN/LRScheduler.cs

```diff
@@ -26,16 +26,18 @@ protected LRScheduler()
 
                     }
 
-                    protected LRScheduler(ILearningRateController optimizer, int last_epoch = -1, bool verbose = false)
+                    protected LRScheduler(Optimizer optimizer, int last_epoch = -1, bool verbose = false)
```

Comment:
It would seem more natural to have `LRScheduler` work per parameter group, not per optimizer. Same for all other schedulers.

Reply:
Following the Pytorch API.

https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.StepLR.html#steplr

 The schedulers get the parameter groups from the optimizer.

Reply:
Is there a reason for the scheduler to store a link to the optimizer? If it only needs to store parameter groups internally, this it what is should it. You can have a constructor overload taking optimizer.

Reply:
Sounds like a pretty good explanation to me.

But if user can remove a parameter group, the corresponding state should be removed as well.